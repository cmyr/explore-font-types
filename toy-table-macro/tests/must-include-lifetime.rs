toy_table_macro::tables! {
// this should fail because it needs to have a lifetime
    MyTable {
        item_count: Uint24,
        #[count(item_count)]
        items: [Uint24],
    }
}


toy_table_macro::tables! {
// this should fail because it needs to *not* a lifetime
    MyTable<'a> {
        item_count: Uint24,
        item_size: Int16,

    }
}

fn main() {}
