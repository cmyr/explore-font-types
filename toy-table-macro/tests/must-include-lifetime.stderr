error: object containing array or offset requires lifetime param (MyTable<'a>)
 --> tests/must-include-lifetime.rs:3:5
  |
3 |     MyTable {
  |     ^^^^^^^

error: only objects containing arrays or offsets require lifetime
  --> tests/must-include-lifetime.rs:13:13
   |
13 |     MyTable<'a> {
   |             ^^
